railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -231,6 +231,24 @@ def initialize(*args)
           raise Error, "Invalid value for --database option. Supported for preconfiguration are: #{DATABASES.join(", ")}."
         end
 
+        # Fail hard if node or yarn isn't detected on the system
+        if options[:webpack]
+          check_node
+          check_yarn
+        end
+
+        # --webpack uses webpacker, which uses action view helpers
+        # --api skips action view
+        if options[:webpack] && options[:api]
```

Comment:
Where webpacker uses action view? One of the uses cases for webpacker is exacly API applications that want to use client side frameworks.